src/app/container/state/refresh-wizard.service.ts

```diff
@@ -0,0 +1,120 @@
+import { HttpErrorResponse } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { MatSnackBar } from '@angular/material';
+import { transaction } from '@datorama/akita';
+import { MytoolsService } from 'app/mytools/mytools.service';
+import { TokenSource } from 'app/shared/enum/token-source.enum';
+import { UsersService } from 'app/shared/openapi';
+import { SessionQuery } from 'app/shared/session/session.query';
+import { SessionService } from 'app/shared/session/session.service';
+import { UsersService as SwaggerUsersService } from 'app/shared/swagger';
+import { UserQuery } from 'app/shared/user/user.query';
+import { finalize } from 'rxjs/operators';
+import { RefreshWizardQuery } from './refresh-wizard.query';
+import { RefreshWizardStore } from './refresh-wizard.store';
+
+@Injectable()
+export class RefreshWizardService {
+  constructor(
+    private refreshWizardStore: RefreshWizardStore,
+    private dockerRegistriesService: UsersService,
+    private swaggerUsersService: SwaggerUsersService,
+    private userQuery: UserQuery,
+    private refreshWizardQuery: RefreshWizardQuery,
+    private matSnackBar: MatSnackBar,
+    private thing: MytoolsService,
+    private sessionQuery: SessionQuery,
+    private sessionService: SessionService
+  ) {}
+  getOrganizations(dockerRegistry: string) {
+    this.refreshWizardStore.setLoading(true);
+    this.dockerRegistriesService
+      .getDockerRegistriesOrganization(dockerRegistry)
+      .pipe(finalize(() => this.refreshWizardStore.setLoading(false)))
+      .subscribe(
+        organizations => {
+          this.setOrganizations(organizations);
+        },
+        error => {
+          this.setError(error);
+        }
+      );
+  }
+
+  setError(error: HttpErrorResponse | null) {
+    this.refreshWizardStore.update(state => {
+      return {
+        ...state,
+        error: error
+      };
+    });
+  }
+
+  refreshOrganization(organization: string) {
+    this.refreshWizardStore.setLoading(true);
+    this.swaggerUsersService
+      .refreshToolsByOrganization(this.userQuery.getValue().user.id, organization, TokenSource.QUAY)
+      .pipe(finalize(() => this.refreshWizardStore.setLoading(false)))
+      .subscribe(thing => {
+        console.log('I have no idea what this returns');
```

Comment:
I don't understand this method. You're refreshing tools for an org, but then not looking for the result. In the next refreshRepository method, you do call `getMyEntries` after it's been refreshed. Should you be doing that here?

Reply:
This is currently unused, not fully implemented. I guess I'll remove this for now.